rcl/src/rcl/init.c

```diff
@@ -155,13 +156,74 @@ rcl_init(
 
   rmw_localhost_only_t * localhost_only =
     &context->impl->init_options.impl->rmw_init_options.localhost_only;
-  if (RMW_LOCALHOST_ONLY_DEFAULT == *localhost_only) {
+  if (RMW_LOCALHOST_ONLY_DEFAULT != *localhost_only) {
+    RCUTILS_LOG_WARN_NAMED(
+      ROS_PACKAGE_NAME,
+      "'localhost_only' init option is deprecated. Use 'automatic_discovery_range' and "
+      "'static_peers' instead.");
+  } else {
     // Get actual localhost_only value based on environment variable, if needed.
     ret = rcl_get_localhost_only(localhost_only);
     if (RCL_RET_OK != ret) {
       fail_ret = ret;
       goto fail;
     }
+    if (RMW_LOCALHOST_ONLY_DEFAULT != *localhost_only) {
+      RCUTILS_LOG_WARN_NAMED(
+        ROS_PACKAGE_NAME,
+        "ROS_LOCALHOST_ONLY is deprecated. Use ROS_AUTOMATIC_DISCOVERY_RANGE_DEFAULT and "
+        "ROS_STATIC_PEERS instead.");
+    }
+  }
+
+  rmw_discovery_options_t * discovery_options =
+    &context->impl->init_options.impl->rmw_init_options.discovery_options;
+  if (RMW_AUTOMATIC_DISCOVERY_RANGE_DEFAULT == discovery_options->automatic_discovery_range) {
+    // Get actual multicast discovery value based on environment variable, if needed
+    ret = rcl_get_discovery_automatic_range(discovery_options);
+    if (RCL_RET_OK != ret) {
+      fail_ret = ret;
+      goto fail;
+    }
+  }
+
+  if (0 == discovery_options->static_peers_count &&
+    discovery_options->automatic_discovery_range != RMW_AUTOMATIC_DISCOVERY_RANGE_OFF)
+  {
+    // Get static peers.
+    // If off is set, it makes sense to not get any static peers.
+    ret = rcl_get_discovery_static_peers(discovery_options, &allocator);
+    if (RCL_RET_OK != ret) {
+      fail_ret = ret;
+      goto fail;
+    }
+  }
+
+  if (discovery_options->static_peers_count > 0 &&
+    discovery_options->automatic_discovery_range == RMW_AUTOMATIC_DISCOVERY_RANGE_OFF)
+  {
+    RCUTILS_LOG_WARN_NAMED(
+      ROS_PACKAGE_NAME,
+      "Note: ROS_AUTOMATIC_DISCOVERY_RANGE is set to OFF, but "
+      "found static peers in ROS_STATIC_PEERS. "
+      "Your settings ROS_STATIC_PEERS will be ignored.");
+  }
+
+  char discovery_range_string[41];  // TODO(gbiggs) Is constexpr available in C11?
+  rcl_automatic_discovery_range_to_string(discovery_range_string, 41, discovery_options);
+  RCUTILS_LOG_INFO_NAMED(
+    ROS_PACKAGE_NAME,
+    "Automatic discovery range is %s",
+    discovery_range_string);
+  RCUTILS_LOG_INFO_NAMED(
+    ROS_PACKAGE_NAME,
+    "Static peers count is %lu",
+    discovery_options->static_peers_count);
+
+  for (size_t ii = 0; ii < discovery_options->static_peers_count; ++ii) {
+    RCUTILS_LOG_INFO_NAMED(
+      ROS_PACKAGE_NAME,
+      "\t%s", discovery_options->static_peers[ii].peer_address);
```

Comment:
I'm not sure about logging these to info every single time. Imagine you've set some of these settings and you're running 100 processes that use ROS in a launch file. The amount of stdout traffic would be huge and redundant. I think it would probably be better to have these as DEBUG statements, but I'd like to hear from others.

Reply:
fixed in df48726